### DISCUSSION TITLE
Finite discrete element method in MOOSE

### DISCUSSION THREAD

**Discussion by caowen1510957** [idaholab/moose#21898] [Q&A General] [answered]
Hello, 

1. Is there a Module for the Finite discrete element method in MOOSE?
https://en.wikipedia.org/wiki/Discrete_element_method

2. For fracturing modeling, phase-field, and discrete element method, can anyone provide a good Pro and Con for these two methods?

3. Or is there a "better (advanced)" method adapted in MOOSE for fracturing induced by dynamic loading?

**Comment by GiudGiud** [comment]
@recuero

**Comment by jiangwen84** [comment, marked as answer]
We do not have DEM implemented in MOOSE.  For fracture modeling,  we have XFEM, PF and Peridynamics.  You should be able to find literatures to discuss their pro and cons in details.

**Reply by caowen1510957** [reply]
> We do not have DEM implemented in MOOSE. For fracture modeling, we have XFEM, PF and Peridynamics. You should be able to find literatures to discuss their pro and cons in details.

Thanks, the following question, for the boundary, is there an option to add an absorbing boundary?

**Reply by aaelmeli** [reply]
> > We do not have DEM implemented in MOOSE. For fracture modeling, we have XFEM, PF and Peridynamics. You should be able to find literatures to discuss their pro and cons in details.
> 
> Thanks, the following question, for the boundary, is there an option to add an absorbing boundary?

There is a way to apply simple absorbing boundary conditions, check this  #20850. Also, a Robin-kind boundary conditions is available for electromagnetics, but I think - I have not tried it though- it may be easily extended to elastodynamics, check this https://mooseframework.inl.gov/moose/source/bcs/EMRobinBC.html and https://mooseframework.inl.gov/moose/source/bcs/VectorTransientAbsorbingBC.html.
